docs/src/protocol_library.md

```diff
@@ -0,0 +1,127 @@
+| `get_recipient` | Returns the [recipient](note.md#note-recipient-restricting-consumption) of the input note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[RECIPIENT]` | Any |
+| `get_metadata` | Returns the [metadata](note.md#metadata) of the input note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[METADATA]` | Any |
+
+## Output Note Procedures (`miden::output_note`)
+
+Output note procedures can be used to fetch data on output notes created by the transaction.
+
+| Procedure | Description | Context |
+| --- | --- | --- |
+| `get_assets_info` | Returns the information about assets in the output note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[ASSETS_COMMITMENT, num_assets]` | Any |
+| `get_assets` | Writes the assets of the output note with the specified index into memory starting at the specified address.<br><br>Inputs: `[dest_ptr, note_index]`<br>Outputs: `[num_assets, dest_ptr, note_index]` | Any |
+| `get_recipient` | Returns the [recipient](note.md#note-recipient-restricting-consumption) of the output note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[RECIPIENT]` | Any |
+| `get_metadata` | Returns the [metadata](note.md#metadata) of the output note with the specified index.<br><br>Inputs: `[note_index]`<br>Outputs: `[METADATA]` | Any |
+
+## Transaction Procedures (`miden::tx`)
+
+Transaction procedures manage transaction-level operations including note creation, context switching, and reading transaction metadata.
+
+| Procedure | Description | Context |
+| --- | --- | --- |
+| `get_block_number` | Returns the block number of the transaction reference block.<br><br>Inputs: `[]`<br>Outputs: `[num]` | Any |
+| `get_block_commitment` | Returns the block commitment of the reference block.<br><br>Inputs: `[]`<br>Outputs: `[BLOCK_COMMITMENT]` | Any |
+| `get_block_timestamp` | Returns the timestamp of the reference block for this transaction.<br><br>Inputs: `[]`<br>Outputs: `[timestamp]` | Any |
+| `get_input_notes_commitment` | Returns the input notes commitment hash.<br><br>Inputs: `[]`<br>Outputs: `[INPUT_NOTES_COMMITMENT]` | Any |
+| `get_output_notes_commitment` | Returns the output notes commitment hash.<br><br>Inputs: `[]`<br>Outputs: `[OUTPUT_NOTES_COMMITMENT]` | Any |
+| `get_num_input_notes` | Returns the total number of input notes consumed by this transaction.<br><br>Inputs: `[]`<br>Outputs: `[num_input_notes]` | Any |
+| `get_num_output_notes` | Returns the current number of output notes created in this transaction.<br><br>Inputs: `[]`<br>Outputs: `[num_output_notes]` | Any |
+| `create_note` | Creates a new note and returns the index of the note.<br><br>Inputs: `[tag, aux, note_type, execution_hint, RECIPIENT]`<br>Outputs: `[note_idx]` | Native & Account |
+| `add_asset_to_note` | Adds the ASSET to the note specified by the index.<br><br>Inputs: `[ASSET, note_idx]`<br>Outputs: `[ASSET, note_idx]` | Native |
```

Comment:
Is the context correct here? I thought we could call these from note scripts as well. If so, the context should probably be something like:
```
Native & Account, Note
```

Reply:
Makes sense - though, I wonder if we can improve these definitions (in a future PR) because even I got confused a bit (I thought this implied that we can't call these from `Note` context - but we actually can).